### PR TITLE
Add getOrElseF method to XorT

### PR DESCRIPTION
**Pull request by easel**

### CODE REVIEW

**Comment by easel** [comment]
@ceedubs per gitter discussion, here's a first attempt at an implementation + test for XorT.getOrElseF

**Comment by adelbertc** [comment]
:+1:

File: core/src/main/scala/cats/data/XorT.scala [hunk 1]
```diff
@@ -22,6 +22,13 @@ final case class XorT[F[_], A, B](value: F[A Xor B]) {
 
   def getOrElse[BB >: B](default: => BB)(implicit F: Functor[F]): F[BB] = F.map(value)(_.getOrElse(default))
 
+  def getOrElseF[BB >: B](default: => F[BB])(implicit F: Monad[F]): F[BB] = {
+    F.flatMap(value) {
+      case Xor.Left(a) => default
+      case Xor.Right(a) => F.pure(a)
```
Comment (ceedubs) [inline core/src/main/scala/cats/data/XorT.scala]: Extremely minor: I'd be inclined to use the following names for the captured values:

```scala
case Xor.Left(_) => default
case Xor.Right(b) => F.pure(b)
```

As this doesn't actually result in any difference, I'm not sure I can really justify the nitpick though...

**Comment by ceedubs** [comment]
:+1:

**Comment by easel** [comment]
Easy enough to change. Update incoming.

**Comment by easel** [comment]
I'm pretending I didn't read the contributing file and force pushing the change anyway. Match parameters now updated per @ceedubs suggestion.

**Comment by ceedubs** [comment]
Thanks!

**Comment by easel** [comment]
Cool, thanks for the help!

**Pull request merged**